cloudflare.go

```diff
@@ -160,29 +160,28 @@ func (api *API) makeRequestWithAuthType(ctx context.Context, method, uri string,
 
 func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, uri string, params interface{}, authType int, headers http.Header) ([]byte, error) {
 	// Replace nil with a JSON object if needed
-	var jsonBody []byte
+	var reqBody io.Reader
 	var err error
-	if params != nil {
-		if paramBytes, ok := params.([]byte); ok {
-			jsonBody = paramBytes
-		} else {
-			jsonBody, err = json.Marshal(params)
-			if err != nil {
-				return nil, errors.Wrap(err, "error marshalling params to JSON")
-			}
+
+	switch p := params.(type) {
+	case []byte:
+		reqBody = bytes.NewReader(p)
+	case io.Reader:
+		reqBody = p
+	case nil:
+		// empty request body
+	default:
+		jsonBody, err := json.Marshal(p)
+		if err != nil {
+			return nil, errors.Wrap(err, "error marshalling params to JSON")
```

Comment:
do we have a `errMarshalError`?

Reply:
I'm not sure, checking.

Reply:
Not that I can find. This is a direct copy of the error that this library already returned.